version: "3"

services:
  db:
    image: postgres:16-alpine
    restart: on-failure
    env_file:
      - .env
    expose:
      - "5432"
    volumes:
      - pg_data:/var/lib/postgresql/test_data
    healthcheck:
      test: ["CMD-SHELL", "-c", "pg_isready -U $POSTGRES_USER"]
      interval: 10s
      retries: 3
      timeout: 5s

  app:
    build: .
    tty: true
    ports:
      - "8000:8000"
    command: sh -c "python manage.py migrate && python manage.py search_index --rebuild && python manage.py csu && python manage.py runserver 0.0.0.0:8000"
    depends_on:
      db:
        condition: service_healthy
      elasticsearch:
        condition: service_healthy
    volumes:
      - .:/app
    env_file:
      - .env
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:8.31.0
    environment:
      - discovery.type=single-node
      - xpack.security.enabled=false
    volumes:
      - esdata:/usr/share/elasticsearch/test_data
    ports:
      - "9200:9200"


volumes:
  pg_data:
  esdata: